YSMOD-565

lurajon opened the pull request:
- la til mapping fra graphql til JPA sortering

lurajon committed:
YSMOD-565

- la til mapping fra graphql til JPA sortering

erlendkri commented on src/main/kotlin/no/nav/yrkesskade/saksbehandling/graphql/common/model/Page.kt:
```diff
@@ -4,9 +4,17 @@ import org.springframework.data.domain.PageRequest
-        return PageRequest.of(page, size, Sort.by(
-            Sort.Direction.ASC, "opprettetTidspunkt"))
+        val direction = if (sortDirection != null) Sort.Direction.fromString(sortDirection.name) else Sort.Direction.ASC
+        val field = sortField?.name ?: "opprettetTidspunkt"
```
bruke enumen sortFieldType?

erlendkri replied:
Og er class Page til generelt bruk? Vil den alltid ha en kolonne opprettetTidspunkt?

lurajon replied:
Bra sett! Jeg tenkte også på dette når jeg var på vei hjem. Page høres generell ut, men mulig den må være spesifikk. Det samme vil skje med sortFieldType, da vi kun skal kunne støtte verdier som faktisk finnes i tabellen. Jeg antar at jeg kommer til å flytte Sort ut av Page, slik at page blir generell, mens Sort blir spesifikk.

erlendkri reviewed (commented):
Nydelig! En liten kommentar...

lurajon committed:
YSMOD-565

- endret til en litt mer generisk utførelse av Page og legger på Sortering som ett spesifikk argument

lurajon committed:
YSMOD-565

- oppdatert med master

erlendkri reviewed (approved):
Supert!
